database/models/files.db.php

```diff
@@ -51,4 +51,10 @@ static function get($file_id): SharedFile|null
         return em()->find(SharedFile::class, $file_id);
     }
 
+    /** @return self[] */
+    static function findBy($permission): array
+    {
+        return em()->getRepository(self::class)->findBy(['permission_level' => $permission]);
+    }
```

Comment:
extracting this logic allows to have type safety! Then you get IDE suggestions from the SharedFile class.

I believe there is some way to get generics to work with Psalm... maybe we should look into it 🤔